lib/utils.js

```diff
@@ -4,60 +4,61 @@ import { exec } from 'teen_process';
 import { waitForCondition } from 'asyncbox';
 import { getVersion } from 'appium-xcode';
 import { getDevices } from 'node-simctl';
-import { fs, system } from 'appium-support';
+import { fs } from 'appium-support';
 import { Certificate } from './certificate';
 import path from 'path';
 import Simulator from './simulator-xcode-6';
 
 
-const OSASCRIPT_TIMEOUT = 10000;
+const DEFAULT_SIM_SHUTDOWN_TIMEOUT = 10000;
 
-async function sendKillAll (appName) {
+// pgrep/pkill exit codes:
+// 0       One or more processes were matched.
+// 1       No processes were matched.
+// 2       Invalid options were specified on the command line.
+// 3       An internal error occurred.
+
+async function killApp (appName, isForceKill = false) {
+  let args = ['-x', appName];
+  if (isForceKill) {
+    args.splice(0, 0, '-9');
+  }
   try {
-    await exec('killall', [appName]);
-  } catch (err) {
-    // if no process exists we'll get an error that we expect.
-    // otherwise throw it again
-    if (err.stderr.indexOf(`No matching processes belonging to you were found`) === -1) {
-      throw err;
-    }
+    await exec('pkill', args);
+    return 0;
+  } catch (e) {
+    if (!_.isUndefined(e.code)) return e.code;
+    log.error(`Got unexpected error while trying to kill ${appName}`);
```

Comment:
not to nitpick, but it would be nice to know what the error is in this case:

```js
log.errorAndThrow(`Unexpected error while trying to kill ${appName}: ${e.message}`);
```